en/editor/publish/custom-build-plugin.md

```diff
@@ -44,92 +54,124 @@ export const configs: IConfigs = {
-    class?: string | string[]; // 需要设置在当前 ui-prop 上的样式名称
+    class?: string | string[]; // The name of the style that needs to be set on the current ui-prop
 }
 ```
 
-其中 `IOptionsBase` 的接口定义需要参考 [ui-prop 自动渲染规则定义](../extension/ui.md)。
+For the interface definition of `IOptionsBase` please refer to [ui-prop automatic rendering rule definition](../extension/ui.md).
+
+## Custom build hook function code configuration
 
-## 自定义构建钩子函数代码配置
+In the script module defined by the hooks field in the entry configuration, hook functionss can be written that build the life cycle. In different hook functions, the data received will be different. All hook functions run in the build process, and the engine method can be used directly in the build process. If you need to use `Editor`, adding the code `import * as Editor from'editor';` manually is required.
 
-入口配置里的 hooks 字段定义的脚本模块内可以编写构建生命周期的钩子函数，在不同的钩子函数内部，接受到的数据会有差异。钩子函数全部都运行在构建进程内，在构建进程内可以直接使用引擎方法，如需使用 `Editor` 需要添加代码 `import * as Editor from 'editor';` 手动 require，关于 Editor 的接口介绍还请参考编辑器的插件开发文档。公开的钩子函数与构建的生命周期的关系可以参考下图：
+The relationship between the public hook function and the life cycle of the build can be seen in the following figure:
 
 ![build-process](./custom-project-build-template/build-process.jpg)
 
-钩子函数的大致接口定义如下图：
+The rough interface definition of hook function is as follows:
 
 ```ts
 declare interface IHook {
-    throwError?: boolean; // 插件注入的钩子函数，在执行失败时是否直接退出构建流程
-    // ------------------ 钩子函数 --------------------------
+    throwError?: boolean; // Whether the hook function injected by the plug-in directly exits the build process when the execution fails
+    // ------------------ hook function --------------------------
     onBeforeBuild?: IBaseHooks;
     onBeforeCompressSettings?: IBaseHooks;
     onAfterCompressSettings?: IBaseHooks;
     onAfterBuild?: IBaseHooks;
+
+    // Compile the generated hook function (only valid when the platform is built with a "generation" process)
+    onBeforeMake?: (root: string, options: IBuildTaskOptions) => void;
+    onAfterMake?: (root: string, options: IBuildTaskOptions) => void;
 }
 type IBaseHooks = (options: IBuildTaskOptions, result?: IBuildResult) => void;
 ```
 
-> **注意**：在 `onBeforeCompressSettings` 开始才能访问到 `result` 参数，并且传递到钩子函数内的 `options` 是实际构建进程中使用 `options` 一个副本仅作为信息的获取参考，因而直接修改它并不会真正的影响构建。构建参数的修改请使用入口的 `options` 来配置。由于接口定义众多，详细的接口定义可以参考构建插件模板文件夹内的 `@types/builder.d.ts` 文件。
+> **Note**: the `result` parameter can be accessed at the beginning of `onBeforeCompressSettings`, and the `options` passed to the hook function is used in the actual build process. A copy of `options` is only used as a reference for information acquisition, so directly Modifying it does not really affect the build. To modify the build parameters, please use the `options` of the entry to configure. Due to the numerous interface definitions, you can refer to the `@types/builder.d.ts` file in the build plugin template folder for detailed interface definitions.
```

Comment:
```suggestion
> **Note**: the `result` parameter can be accessed only at the beginning of `onBeforeCompressSettings`, and the `options` passed to the hook function is a copy of the `options` used in the actual build process, and only used as a reference for information acquisition, so directly modifying it does not really affect the build process, although it can be modified successfully. To modify the build parameters, please set in the `options` field of the entry configuration code. Due to the numerous interface definitions, you can refer to the @types/packages/builder folder in the build extension package for detailed interface definitions.
```